build/Microsoft.DotNet.Cli.Prepare.targets

```diff
@@ -102,11 +107,80 @@
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_DownloadAndExtractItem Include="CombinedSharedHostAndFrameworkArchive"
+                               Condition="!Exists('$(CombinedSharedHostAndFrameworkArchive)')">
+        <Url>$(SharedFrameworkArchiveBlobRootUrl)/$(CombinedFrameworkHostCompressedFileName)</Url>
+        <DownloadFileName>$(CombinedSharedHostAndFrameworkArchive)</DownloadFileName>
+        <ExtractDestintation>$(SharedFrameworkPublishDirectory)</ExtractDestintation>
+      </_DownloadAndExtractItem>
+
+      <_DownloadAndExtractItem Include="SharedFrameworkInstallerFile"
+                               Condition="!Exists('$(SharedFrameworkInstallerFile)')">
+        <Url>$(CoreSetupInstallerBlobRootUrl)/$(SharedFrameworkVersion)/$(SharedFrameworkInstallerFileName)</Url>
+        <DownloadFileName>$(SharedFrameworkInstallerFile)</DownloadFileName>
+        <ExtractDestintation></ExtractDestintation>
+      </_DownloadAndExtractItem>
+
+      <_DownloadAndExtractItem Include="SharedHostInstallerFile"
+                               Condition="!Exists('$(SharedHostInstallerFile)')">
+        <Url>$(CoreSetupInstallerBlobRootUrl)/$(SharedHostVersion)/$(SharedHostInstallerFileName)</Url>
+        <DownloadFileName>$(SharedHostInstallerFile)</DownloadFileName>
+        <ExtractDestintation></ExtractDestintation>
+      </_DownloadAndExtractItem>
+
+      <_DownloadAndExtractItem Include="HostFxrInstallerFile"
+                               Condition="!Exists('$(HostFxrInstallerFile)')">
+        <Url>$(CoreSetupInstallerBlobRootUrl)/$(HostFxrVersion)/$(HostFxrInstallerFileName)</Url>
+        <DownloadFileName>$(HostFxrInstallerFile)</DownloadFileName>
+        <ExtractDestintation></ExtractDestintation>
+      </_DownloadAndExtractItem>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="DownloadHostAndSharedFxArtifacts"
+          DependsOnTargets="SetupDownloadHostAndSharedFxInputsOutputs">
+
+    <DownloadFile Condition=" '@(_DownloadAndExtractItem)' != '' "
+                  Uri="%(_DownloadAndExtractItem.Url)"
+                  DestinationPath="%(_DownloadAndExtractItem.DownloadFileName)" />
+
+    <ExtractArchive Condition=" '%(_DownloadAndExtractItem.ExtractDestintation)' != '' "
+                    InputFile="%(_DownloadAndExtractItem.DownloadFileName)"
+                    DestinationDirectory="%(_DownloadAndExtractItem.ExtractDestintation)" />
+  </Target>
+
+  <Target Name="RestorePackages"
+          DependsOnTargets="Init">
 
     <DotNetRestore WorkingDirectory="$(RepoRoot)/src" ToolPath="$(DotNetPath)" />
     <DotNetRestore WorkingDirectory="$(RepoRoot)/tools" ToolPath="$(DotNetPath)" />
-    
+
+  </Target>
+
+  <Target Name="ZipTemplates">
+    <PropertyGroup>
+      <_DotNetNewFolder>$(RepoRoot)/src/dotnet/commands/dotnet-new</_DotNetNewFolder>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <DotNetNewArchive Include="$([System.IO.Directory]::GetDirectories($(_DotNetNewFolder)))" />
```

Comment:
It does what it normall does in .NET:  https://msdn.microsoft.com/en-us/library/c1sez4sc(v=vs.110).aspx